### PR TITLE
table --collapse dont do truncation return message instead

### PR DESCRIPTION
**Pull request by zhiburt**
Reverts #8042

I've just noticed that #8042 was merged, but I didn't addressed your @fdncred last comment.
This PR reverts #8042 and returns a message in cases where we need truncation/wrapping.

https://github.com/nushell/nushell/blob/157b7e0b603c43d8e4f3bd0b028218d1e1d29114/crates/nu-command/tests/commands/table.rs#L234-L240

### CODE REVIEW

**Comment by fdncred** [comment]
ok, thanks @zhiburt

**Comment by amtoine** [comment]
i though maybe, in case the terminal is not wide enough, still print the `table --collapse` but WITH a clear warning message, e.g.
```bash
> open Cargo.toml | get package.metadata | table -c
╭──────────┬───────────┬────────────────────────┬─────────┬────────────────────────────────────────────╮
│ binstall │ overrides │ x86_64-pc-windows-msvc │ pkg-fmt │                    zip                     │
│          ├───────────┼────────────────────────┴─────────┴────────────────────────────────────────────┤
│          │  pkg-fmt  │                                     tgz                                       │
│          ├───────────┼───────────────────────────────────────────────────────────────────────────────┤
│          │  pkg-url  │ { repo }/releases/download/{ version }/{ name }-{ version }-{ target }.{ arch │
╰──────────┴───────────┴───────────────────────────────────────────────────────────────────────────────╯

WARNING: because you terminal is too narrow (... columns) the output of the previous `table --collapse` command has been truncated on the right!
```

> **Note**
> see the output of `binstall.pkg-url` has been truncated at `... arch`

**Comment by zhiburt** [comment]
> i though maybe, in case the terminal is not wide enough, still print the table --collapse but WITH a clear warning message, e.g.

We could do that but you'd see a broken table above. 
Maybe it makes sense so you'd have no need to guess about the necessary width.

**Comment by amtoine** [comment]
> We could do that but you'd see a broken table above. Maybe it makes sense so you'd have no need to guess about the necessary width.
>
yeah, i think it's better to give a broken table, with a clear and visible comment rather than no output at all :thinking: 
no output because of a terminal not wide enough would force the user to resize his terminal even though the whole data might not be mandatory, e.g. i do not need to see the whole `binstall.pkg-url` in the example i gave above, a partial but existing output is enough
=> i think `table --collapsed` is mainly used to explore or share structured data! as it can not be piped further, it can not be used to compute things, so removing the output altogether sounds a bit strong to me :yum: 

> Or we could expend the comment to
> 
> `"Couldn't fit table into 80 columns (need at least 120)!"`
> 
> We could follow this approach for other table styles as well, I guess it makes sense as we have the value in any way?
>
giving the minimum required width for a complete table is a good idea :+1: 

**TL;DR**: i think always giving an output, especially when it's the end of a pipe, even partially broken, is better than nothing with a clear warning about what happened and possible hints
**TL;DR no2**: maybe this discussion should move to another issue? :wink:

**Comment by amtoine** [comment]
for instance, when i zoom too much in the `nushell` source, the `ls` command truncates the output table and does not remove the whole output
```bash
> ls | first 10
╭───┬─────────────────┬──────┬──────────┬─────╮
│ # │      name       │ type │   size   │ ... │
├───┼─────────────────┼──────┼──────────┼─────┤
│ 0 │ CODE_OF_CONDUCT │ file │   3.4 KB │ ... │
│   │ .md             │      │          │     │
│ 1 │ CONTRIBUTING.md │ file │   3.6 KB │ ... │
│ 2 │ Cargo.lock      │ file │ 148.3 KB │ ... │
│ 3 │ Cargo.toml      │ file │   4.9 KB │ ... │
│ 4 │ Cross.toml      │ file │    363 B │ ... │
│ 5 │ LICENSE         │ file │   1.1 KB │ ... │
│ 6 │ README.md       │ file │  13.3 KB │ ... │
│ 7 │ README.release. │ file │    124 B │ ... │
│   │ txt             │      │          │     │
│ 8 │ assets          │ dir  │   4.1 KB │ ... │
│ 9 │ benches         │ dir  │   4.1 KB │ ... │
╰───┴─────────────────┴──────┴──────────┴─────╯
```
> **Note**
> ```bash
> > term size
> ╭─────────┬────╮
> │ columns │ 47 │
> │ rows    │ 13 │
> ╰─────────┴────╯
> ```

with the `Cargo.toml` example above, we could imagine one of the following output

> **Note**
> below, the command is always `open Cargo.toml | get package.metadata | table -c`

## partial truncation
```bash
╭──────────┬───────────┬────────────────────────┬─────────┬────────────────────────────────────────────────╮
│ binstall │ overrides │ x86_64-pc-windows-msvc │ pkg-fmt │                      zip                       │
│          ├───────────┼────────────────────────┴─────────┴────────────────────────────────────────────────┤
│          │  pkg-fmt  │                                       tgz                                         │
│          ├───────────┼───────────────────────────────────────────────────────────────────────────────────┤
│          │  pkg-url  │ { repo }/releases/download/{ version }/{ name }-{ version }-{ target }.{ arch ... │
╰──────────┴───────────┴───────────────────────────────────────────────────────────────────────────────────╯
```

## full truncation
```bash
╭──────────┬───────────┬────────────────────────┬─────────┬─────╮
│ binstall │ overrides │ x86_64-pc-windows-msvc │ pkg-fmt │ zip │
│          ├───────────┼────────────────────────┴─────────┴─────┤
│          │  pkg-fmt  │                  tgz                   │
│          ├───────────┼────────────────────────────────────────┤
│          │  pkg-url  │                  ...                   │
╰──────────┴───────────┴────────────────────────────────────────╯
```

> **Warning**
> the `...` could probably be highlighted with some colors, or maybe `... (truncated)`? 
> giving the following in the case of the full truncation
> ```bash
> ╭──────────┬───────────┬────────────────────────┬─────────┬─────╮
> │ binstall │ overrides │ x86_64-pc-windows-msvc │ pkg-fmt │ zip │
> │          ├───────────┼────────────────────────┴─────────┴─────┤
> │          │  pkg-fmt  │                  tgz                   │
> │          ├───────────┼────────────────────────────────────────┤
> │          │  pkg-url  │            ... (truncated)             │
> ╰──────────┴───────────┴────────────────────────────────────────╯
> ```

## the message
- with the full truncation above, i think the warning message is not mandatory
- otherwise
```bash
WARNING: because you terminal is too narrow (... columns) the output of the previous `table --collapse` command has been truncated on the right!
```
could be extended with the minimum required size to mitigate all truncations as said above :thumbsup:
